react/features/mobile/audio-mode/middleware.js

```diff
@@ -43,13 +45,13 @@ MiddlewareRegistry.register(({ getState }) => next => action => {
          */
         case CONFERENCE_JOINED:
         case SET_AUDIO_ONLY: {
-            if (getState()['features/base/conference'].conference
-                    || action.conference) {
```

Comment:
I really like the fact that we're not using the action.conference field anymore, because we agreed that Redux is the single source of truth, but I don't see any changes (other than maybe the next invoked earlier) that would make it unnecessary. It (action.conference) was added here with a reason I guess, so although I like that it's not here anymore, I would like to know why it's not needed now.

Reply:
I guess it's about the next being invoked earlier, so that the things can settle, but it's just my guess.

Reply:
The problem was that we are handling both actions in the same handler, so in one case `action.conference` will have a value, but `action.audioOnly` won't, and the other way around. By waiting until `next(action)` runs we guarantee to read the single source of truth (redux) and act accordingly.